### PR TITLE
Issue/486/gps show user position

### PR DESCRIPTION
**Pull request by Bedpaw**
Draft version of handling geolication and showing user position on map

### CODE REVIEW

File: client/src/components/organisms/map-settings-toggler.vue [hunk 1]
```diff
@@ -1,21 +1,25 @@
...
-    class="container"
-    :class="{ 'open': open, 'closed': !open }"
+    class="f-map-settings-toggle-container"
+    :class="{ 'open': open }"
   >
-    <div :class="{'f-display-none': !open}">
+    <div
+      v-touch:swipe.up="() => open = false"
+      class="f-map-settings"
+      :class="{'f-display-none': !open }"
```
Comment (dbetka) [inline client/src/components/organisms/map-settings-toggler.vue:9]: Space

File: client/src/components/organisms/map-settings-toggler.vue [hunk 2]
```diff
@@ -1,21 +1,25 @@
...
+      :class="{ 'opened': open }"
       @click="open = !open"
     >
       <a-icon
-        :name="open ? $icons.names.close : $icons.names.arrow_drop_down"
-        :size="open ? '24' : '36'"
-        :class="{ 'arrow-down': !open, 'close': open }"
         class="arrow-down"
+        :name="open ? $icons.names.expand_less : $icons.names.expand_more"
+        :size="'32'"
```
Comment (dbetka) [inline client/src/components/organisms/map-settings-toggler.vue:22]: Instead `:size="'32'"` use `size="32"`

File: client/src/components/organisms/map-settings-toggler.vue [hunk 3]
```diff
@@ -31,55 +35,11 @@ export default {
   data: () => ({
     open: false,
   }),
+  watch: {
+    open: function (isOpen) {
+      const bannerTimer = document.getElementsByClassName('m-banner-timer')[0];
+      bannerTimer.style.top = isOpen ? '239px' : 0;
```
Comment (dbetka) [inline client/src/components/organisms/map-settings-toggler.vue:41]: Find better solution

File: client/src/components/organisms/map-settings.vue [hunk 4]
```diff
@@ -1,53 +1,64 @@
...
-  <m-input
-    v-if="featuresConfig.lineVisibility"
-    v-model="featuresConfig.lineWidth"
-    placeholder="Szerokość trasy"
-  />
-  <ol>
-    <li
-      v-for="log in $store.state.mapLogs"
-      :key="log"
+  <template v-if="false">
```
Comment (dbetka) [inline client/src/components/organisms/map-settings.vue:12]: To remove before merge

File: client/src/components/organisms/map-settings.vue [hunk 5]
```diff
@@ -1,53 +1,64 @@
...
+    <m-input
+      v-if="featuresConfig.lineVisibility"
+      v-model="featuresConfig.lineWidth"
+      placeholder="Szerokość trasy"
+    />
+  </template>
 
-  <a-button-primary @click="applyChanges">
-    Zapisz zmiany
+  <template v-if="false">
```
Comment (dbetka) [inline client/src/components/organisms/map-settings.vue:42]: To remove before merge

File: client/src/map/controls/geolocation-control.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        communicates.showSuccessTemporary('communicate.geolocation.success', true);
+        map.panTo({ longitude, latitude });
+        break;
+    }
+  }
+
+  static getControl () : HTMLElement {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return map.realMap.getControls().getArray().find((instance: Control) => instance instanceof GeolocationControl).element;
```
Comment (dbetka) [inline client/src/map/controls/geolocation-control.ts:69]: Find better solution

File: client/src/map/index.js [hunk 7]
```diff
@@ -23,7 +26,7 @@ export const map = {
       document.getElementById(elementId)?.firstChild?.remove();
     }
   },
-  panTo ({ latitude, longitude, zoom }) {
+  panTo ({ latitude, longitude, zoom = 16 }) {
```
Comment (dbetka) [inline client/src/map/index.js:29]: Check if current zoom is higher then 16

File: client/src/style/auto-import/components/molecules/banner-timer.sass [hunk 8]
```diff
@@ -2,15 +2,17 @@
...
+  top: 0
+  width: 100%
   display: flex
-  flex-direction: column
 
-  padding: 4px 8px
+  padding: 12px 20px
 
-  background: $color-background-transparent
+  background: white
```
Comment (dbetka) [inline client/src/style/auto-import/components/molecules/banner-timer.sass:11]: Use variables

File: client/src/style/auto-import/components/molecules/countdown-time/countdown-timer-warning.sass [hunk 9]
```diff
@@ -1,2 +1,3 @@
 .m-countdown-timer-warning
-  color: red
+  color: #3FB25D
```
Comment (dbetka) [inline client/src/style/auto-import/components/molecules/countdown-time/countdown-timer-warning.sass:2]: Use variables

File: client/src/style/auto-import/components/organisms/map-settings-toggler.sass [hunk 10]
```diff
@@ -0,0 +1,64 @@
+.f-map-settings-toggle-container
```
Comment (dbetka) [inline client/src/style/auto-import/components/organisms/map-settings-toggler.sass:1]: Do it something with it
`.o-container.f-map-settings-toggler`
`.o-container-map-settings-toggle`

File: client/src/style/auto-import/components/organisms/map-settings-toggler.sass [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+  line-height: 20px
+  text-align: center
+
+  &.open
+    top: 0
+    justify-content: flex-start
+    flex-direction: column
+    align-items: flex-end
+
+.f-map-settings-toggle-button
```
Comment (dbetka) [inline client/src/style/auto-import/components/organisms/map-settings-toggler.sass:23]: `.m-button.f-map-settings-toggle`

File: client/src/style/auto-import/components/organisms/map-settings.sass [hunk 12]
```diff
@@ -0,0 +1,5 @@
+.f-map-settings
```
Comment (dbetka) [inline client/src/style/auto-import/components/organisms/map-settings.sass:1]: .o-map-settings

File: client/src/style/vendors.sass [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+
+body .ol-control
+  background-color: white!important
+  border-radius: 10px!important
+  .ol-zoom-in
+    border-bottom: 1px solid #E6E6E6!important
+  &:hover
+    background-color: transparent
+  button
+    color: #171717
```
Comment (dbetka) [inline client/src/style/vendors.sass:31]: Use variables

File: client/src/style/vendors.sass [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+
+.ol-control button
+  border: none !important
+    radius: 0 !important
+
+body .ol-control
+  background-color: white!important
+  border-radius: 10px!important
+  .ol-zoom-in
+    border-bottom: 1px solid #E6E6E6!important
```
Comment (dbetka) [inline client/src/style/vendors.sass:27]: Use variables

File: client/src/style/vendors.sass [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+.ol-attribution
+  bottom: 20px!important
+
+
+.ol-control button
+  border: none !important
+    radius: 0 !important
+
+body .ol-control
+  background-color: white!important
```
Comment (dbetka) [inline client/src/style/vendors.sass:24]: Use variables

File: client/src/utils/geolocation/geolocation-grade.ts [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+  } else {
+    accuracyEnum = GeoAccuracy.HIGH;
+  }
+
+  return accuracyEnum;
+}
+
+function getLastAccuraciesGrade () {
+  // eslint-disable-next-line no-return-assign
+  setTimeout(() => isInitialized = true, timeToInitialize);
```
Comment (dbetka) [inline client/src/utils/geolocation/geolocation-grade.ts:37]: `setTimeout(() => (isInitialized = true), timeToInitialize);`

File: client/src/utils/geolocation/geolocation.ts [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+
+let lastPositionTime = 0;
+
+function isValidResult () {
+  // only one result per 2 seconds
+  const now = Math.floor(Number(new Date()) / 1000);
+  if (lastPositionTime === now || lastPositionTime === (now - 1)) {
+    return false;
+  }
+  console.log(now);
```
Comment (dbetka) [inline client/src/utils/geolocation/geolocation.ts:27]: remove

File: client/src/utils/geolocation/geolocation.ts [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+): void {
+  if (gameRulesUtils.getRuleValueById(Rules.GeolocationAvailability) === GeolocationAvailabilityOptions.Forbidden) {
+    // TODO handle it somehow in ui when game options ready
+    console.warn('Geolocation not allowed');
+    return;
+  }
+
+  if (trackSub) {
+    return;
+  }
```
Comment (dbetka) [inline client/src/utils/geolocation/geolocation.ts:45]: if (trackSub) return;

**Review by dbetka — CHANGES_REQUESTED** [review]

File: client/src/components/organisms/map-settings.vue (hunk 4 above)
Comment (Bedpaw) [reply client/src/components/organisms/map-settings.vue:12]: done

File: client/src/components/organisms/map-settings.vue (hunk 5 above)
Comment (Bedpaw) [reply client/src/components/organisms/map-settings.vue:42]: done

File: client/src/map/controls/geolocation-control.ts (hunk 6 above)
Comment (Bedpaw) [reply client/src/map/controls/geolocation-control.ts:69]: added TODO

File: client/src/map/index.js (hunk 7 above)
Comment (Bedpaw) [reply client/src/map/index.js:29]: Fixed stange change zoom

File: client/src/style/vendors.sass (hunk 14 above)
Comment (Bedpaw) [reply client/src/style/vendors.sass:27]: $Marble: #e6e6e6

File: client/src/style/vendors.sass (hunk 13 above)
Comment (Bedpaw) [reply client/src/style/vendors.sass:31]: $Dark_Gray: #171717

File: client/src/style/auto-import/components/molecules/banner-timer.sass (hunk 8 above)
Comment (Bedpaw) [reply client/src/style/auto-import/components/molecules/banner-timer.sass:11]: done

File: client/src/style/auto-import/components/molecules/countdown-time/countdown-timer-warning.sass (hunk 9 above)
Comment (Bedpaw) [reply client/src/style/auto-import/components/molecules/countdown-time/countdown-timer-warning.sass:2]: done

File: client/src/style/vendors.sass (hunk 15 above)
Comment (Bedpaw) [reply client/src/style/vendors.sass:24]: done

File: client/src/style/vendors.sass (hunk 14 above)
Comment (Bedpaw) [reply client/src/style/vendors.sass:27]: done

File: client/src/style/vendors.sass (hunk 13 above)
Comment (Bedpaw) [reply client/src/style/vendors.sass:31]: done

File: client/src/style/auto-import/components/organisms/map-settings.sass (hunk 12 above)
Comment (Bedpaw) [reply client/src/style/auto-import/components/organisms/map-settings.sass:1]: done

File: client/src/style/auto-import/components/organisms/map-settings-toggler.sass (hunk 10 above)
Comment (Bedpaw) [reply client/src/style/auto-import/components/organisms/map-settings-toggler.sass:1]: Changed to o-map-settings-toggle

File: client/src/style/auto-import/components/organisms/map-settings-toggler.sass (hunk 11 above)
Comment (Bedpaw) [reply client/src/style/auto-import/components/organisms/map-settings-toggler.sass:23]: changed to m-map-settings-toggle-button

File: client/src/components/organisms/map-settings-toggler.vue (hunk 1 above)
Comment (Bedpaw) [reply client/src/components/organisms/map-settings-toggler.vue:9]: done

**Review by dbetka — APPROVED** [review]

**Pull request merged**